cf/commands/domain/domains_test.go

```diff
@@ -1,137 +1,379 @@
+				Expect(err).NotTo(HaveOccurred())
+				Expect(factory.NewLoginRequirementCallCount()).To(Equal(1))
+				Expect(actualRequirements).To(ContainElement(loginRequirement))
+			})
+
+			It("returns a TargetedOrgRequirement", func() {
+				actualRequirements, err := cmd.Requirements(factory, flagContext)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(factory.NewTargetedOrgRequirementCallCount()).To(Equal(1))
+				Expect(actualRequirements).To(ContainElement(targetedOrgRequirement))
+			})
 
-			Expect(runCommand()).To(BeFalse())
+			It("does not return a RoutingAPIRequirement", func() {
+				actualRequirements, err := cmd.Requirements(factory, flagContext)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(factory.NewRoutingAPIRequirementCallCount()).To(Equal(1))
+				Expect(actualRequirements).NotTo(ContainElement(routingApiRequirement))
+			})
 		})
+	})
 
-		It("fails with usage when invoked with any args what so ever ", func() {
-			requirementsFactory.LoginSuccess = true
-			requirementsFactory.TargetedOrgSuccess = true
+	Describe("Execute", func() {
 
-			runCommand("whoops")
+		It("prints getting domains", func() {
+			cmd.Execute(flagContext)
 			Expect(ui.Outputs).To(ContainSubstrings(
-				[]string{"Incorrect Usage", "No argument required"},
+				[]string{"Getting domains in org my-org"},
 			))
 		})
-	})
 
-	Context("when logged in and an org is targeted", func() {
-		BeforeEach(func() {
-			requirementsFactory.LoginSuccess = true
-			requirementsFactory.TargetedOrgSuccess = true
-			requirementsFactory.OrganizationFields = models.OrganizationFields{
-				Name: "my-org",
-				Guid: "my-org-guid",
-			}
+		It("tries to get the list of domains for org", func() {
+			cmd.Execute(flagContext)
+			Expect(domainRepo.ListDomainsForOrgCallCount()).To(Equal(1))
+			orgGuid, _ := domainRepo.ListDomainsForOrgArgsForCall(0)
+			Expect(orgGuid).To(Equal("my-org-guid"))
 		})
 
-		Context("when there is at least one domain", func() {
+		Context("when list domans for org returns error", func() {
 			BeforeEach(func() {
-				domainRepo.ListDomainsForOrgStub = func(orgGuid string, cb func(models.DomainFields) bool) error {
-					cb(models.DomainFields{
-						Shared: false,
-						Name:   "Private-domain1",
-					})
+				domainRepo.ListDomainsForOrgReturns(errors.New("org-domain-err"))
+			})
 
-					cb(models.DomainFields{
-						Shared: true,
-						Name:   "The-shared-domain",
-					})
+			It("fails with message", func() {
+				Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"FAILED"},
+					[]string{"Failed fetching domains."},
+					[]string{"org-domain-err"},
+				))
+			})
+		})
 
-					cb(models.DomainFields{
-						Shared: false,
-						Name:   "Private-domain2",
-					})
+		Context("when there are no domains for org", func() {
+			BeforeEach(func() {
+				domainFields = []models.DomainFields{}
+				domainRepo.ListDomainsForOrgStub = callBackFunc
```

Comment:
This callBackFunc doesn't need to be a top-level var. I would replace the assignment of callBackFunc in the top-level BeforeEach with line 185 here. Just assign it.